provider/helm3/implementer.go

```diff
@@ -0,0 +1,138 @@
+package helm3
+
+import (
+    "os"
+    "strings"
+
+    "helm.sh/helm/v3/pkg/chart"
+
+	log "github.com/sirupsen/logrus"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+    // "helm.sh/helm/v3/pkg/cli"
+
+    "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// to do:
+	// * update to latest chart package
+	// * udpate the paramateres for the function
+
+const DefaultUpdateTimeout = 300
+
+// Implementer - generic helm implementer used to abstract actual implementation
+type Implementer interface {
+	// ListReleases(opts ...helm.ReleaseListOption) ([]*release.Release, error)
+    ListReleases() ([]*release.Release, error)
+	UpdateReleaseFromChart(rlsName string, chart *chart.Chart, vals map[string]string, namespace string, opts ...bool) (*release.Release, error)
+}
+
+// Helm3Implementer - actual helm3 implementer
+type Helm3Implementer struct {
+	// actionConfig *action.Configuration
+    HelmDriver string
+    KubeContext string
+    KubeToken string
+    KubeAPIServer string
+}
+
+// NewHelm3Implementer - get new helm implementer
+func NewHelm3Implementer() *Helm3Implementer {
+    // settings := cli.New()
+
+    // actionConfig := &action.Configuration{}
+    // // You can pass an empty string instead of settings.Namespace() to list
+    // // all namespaces
+    // if err := actionConfig.Init(settings.RESTClientGetter(), "", os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+    //     log.Printf("%+v", err)
+    //     os.Exit(1)
+    // }
+    return &Helm3Implementer{
+        HelmDriver:       os.Getenv("HELM_DRIVER"),
+        KubeContext:      os.Getenv("HELM_KUBECONTEXT"),
+        KubeToken:        os.Getenv("HELM_KUBETOKEN"),
```

Comment:
what sets these values?

Reply:
I’m not entirely sure what sets these values, but they’re the same values from when we were making a call to cli.New() to setup the action configuration (as seen here https://github.com/helm/helm/blob/v3.2.3/pkg/cli/environment.go#L65-L79).
I’m still trying to verify this, but I’m guessing helm sets these values when you deploy a chart and uses values corresponding with the output of `helm env`.